### PR TITLE
Add filters to productVariants query

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because adding filters to `productVariants` query

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1253 Add SKU and name to Variants search](https://app.clickup.com/t/2549495/SALEOR-1253)

File: saleor/graphql/product/filters.py [hunk 1]
```diff
@@ -336,6 +342,17 @@ class Meta:
...
 
+class ProductVariantFilter(django_filters.FilterSet):
+    search = django_filters.CharFilter(
+        method=filter_fields_containing_value("name", "product__name", "sku")
+    )
+    sku = ListObjectTypeFilter(input_class=graphene.String, method=filter_sku_list)
+
+    class Meta:
+        model = ProductVariant
+        fields = ["search"]
```
Comment (korycins) [inline saleor/graphql/product/filters.py:353]: `fields = ["search", 'sku']` ?

File: saleor/graphql/product/schema.py [hunk 2]
```diff
@@ -296,8 +302,12 @@ def resolve_product_type(self, info, id):
     def resolve_product_types(self, info, **kwargs):
         return resolve_product_types(info, **kwargs)
 
-    def resolve_product_variant(self, info, id):
-        return graphene.Node.get_node_from_global_id(info, id, ProductVariant)
+    def resolve_product_variant(self, info, id=None, sku=None):
+        validate_one_of_args_is_in_query("id", id, "sku", sku)
+        if id:
+            return graphene.Node.get_node_from_global_id(info, id, ProductVariant)
+        if sku:
```
Comment (korycins) [inline saleor/graphql/product/schema.py:309]: I assume that `validate_one_of_args_is_in_query` already checked if skur or id is here.  So second if is not needed 
```
validate_one_of_args_is_in_query("id", id, "sku", sku)
if id:
    return graphene.Node.get_node_from_global_id(info, id, ProductVariant)
 return resolve_product_varaint_by_sku(info, sku=sku)
```

**Review by korycins — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/product/resolvers.py [hunk 3]
```diff
@@ -62,6 +62,18 @@ def resolve_product_types(info, **_kwargs):
     return models.ProductType.objects.all()
 
 
+def resolve_product_varaint_by_sku(info, sku):
```
Comment (maarcingebala) [inline saleor/graphql/product/resolvers.py:65]: Typo: varaint

File: saleor/graphql/product/schema.py [hunk 4]
```diff
@@ -226,16 +228,20 @@ class ProductQueries(graphene.ObjectType):
...
     product_variant = graphene.Field(
         ProductVariant,
-        id=graphene.Argument(
-            graphene.ID, description="ID of the product variant.", required=True
+        id=graphene.Argument(graphene.ID, description="ID of the product variant.",),
+        sku=graphene.Argument(
+            graphene.String, description="Sku of the product variant."
         ),
-        description="Look up a product variant by ID.",
+        description="Look up a product variant by ID or sku.",
```
Comment (maarcingebala) [inline saleor/graphql/product/schema.py:235]: :patrys-eye:

```suggestion
        description="Look up a product variant by ID or SKU.",
```

File: saleor/graphql/product/resolvers.py (hunk 3 above)
Comment (korycins) [reply saleor/graphql/product/resolvers.py:65]: 👮

File: saleor/graphql/product/schema.py (hunk 4 above)
Comment (fowczarek) [reply saleor/graphql/product/schema.py:235]: :police_officer:

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**